GithubBrowserSample/app/src/main/java/com/android/example/github/ui/repo/RepoFragment.kt

```diff
@@ -87,23 +89,36 @@ class RepoFragment : Fragment(), Injectable {
             }
         }
         binding = dataBinding
+        sharedElementReturnTransition = TransitionInflater.from(context).inflateTransition(R.transition.move)
         return dataBinding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         repoViewModel = ViewModelProviders.of(this, viewModelFactory)
-            .get(RepoViewModel::class.java)
+                .get(RepoViewModel::class.java)
+        val params = RepoFragmentArgs.fromBundle(arguments!!)
         repoViewModel.setId(params.owner, params.name)
         binding.setLifecycleOwner(viewLifecycleOwner)
         binding.repo = repoViewModel.repo
 
-        val adapter = ContributorAdapter(dataBindingComponent, appExecutors) { contributor ->
+        val adapter = ContributorAdapter(dataBindingComponent, appExecutors) {
+            contributor, imageView ->
+            val extras = FragmentNavigatorExtras(
+                    imageView to contributor.login
+            )
             navController().navigate(
-                RepoFragmentDirections.showUser(contributor.login)
+                    RepoFragmentDirections.showUser(contributor.login, contributor.avatarUrl ?: ""),
+                    extras
             )
         }
         this.adapter = adapter
         binding.contributorList.adapter = adapter
+        postponeEnterTransition()
+        binding.contributorList.getViewTreeObserver()
+                .addOnPreDrawListener {
+                    startPostponedEnterTransition()
+                    true
```

Comment:
i think you need to remove the pre-draw listener here ?

Reply:
It's a good practice.  No SAM Lambda for me.